node-metrics/src/lib.rs

```
@@ -240,20 +244,43 @@ pub async fn run_standalone_service(options: Options) {
         },
     }
 
-    let (leaf_sender, leaf_receiver) = mpsc::channel(10);
+    let (leaf_and_block_pair_sender, leaf_and_block_pair_receiver) = mpsc::channel(10);
+
+    let client = surf_disco::Client::new(options.leaf_stream_base_url().clone());
+
+    // Let's get the current starting block height.
+    let block_height = {
+        let block_height_result = client.get("status/block-height").send().await;
+        let block_height: u64 = match block_height_result {
+            Ok(block_height) => block_height,
+            Err(err) => {
+                tracing::warn!("retrieve block height request failed: {}", err);
+                panic!("error retrieving block height request failed: {}", err);
```

I'm unsure what depends on this library, but panicking after a single request seems drastic. Should we retry a few times?

---

So looking closer, I guess we can't proceed if this fails, so panicking may be appropriate. But shouldn't we retry a few times?

---

The only thing that depends on this library at the moment is the `main` file that launches it within the same directory:

`node-metrics/src/main.rs`.

I will look into retrying this initial request.  Though this only happens on service launch at the moment.  So if the service isn't able to secure this value, it's better to fail fast and allow for the service to be restarted by the underlying orchestration services, which will effectively retry them anyway.

---

I had imagined that the intent was for the orchestration layer to restart the service. So I think it is *workable* but not ideal. Network issues are frequent and I think a 1 to 1 relationship between network caos and service crashes is problematic. Please consider adding retries in a future update.

---

Oh now I see you already did this in [1f8fb0e](https://github.com/EspressoSystems/espresso-network/pull/2351/commits/1f8fb0ecc02fb93d30e2c4d60c031f46aad8f8c4)